.github/workflows/main.yml

```diff
@@ -251,29 +251,25 @@ jobs:
 #          cd docs/_build/html
 #          mkdir artifacts
 #          tar -zcvf artifacts/tutorials.tar.gz --exclude=./artifacts .
-#        if: ${{ matrix.python-version == 3.8 && !startsWith(github.ref, 'refs/heads/stable') && !startsWith(github.base_ref, 'stable/') }}
+#        if: ${{ matrix.python-version == 3.9 && !startsWith(github.ref, 'refs/heads/stable') && !startsWith(github.base_ref, 'stable/') }}
 #        shell: bash
 #      - name: Run upload stable tutorials
 #        uses: actions/upload-artifact@v4
 #        with:
 #          name: tutorials-stable${{ matrix.python-version }}
 #          path: docs/_build/html/artifacts/tutorials.tar.gz
-#        if: ${{ matrix.python-version == 3.8 && !startsWith(github.ref, 'refs/heads/stable') && !startsWith(github.base_ref, 'stable/') }}
+#        if: ${{ matrix.python-version == 3.9 && !startsWith(github.ref, 'refs/heads/stable') && !startsWith(github.base_ref, 'stable/') }}
   Deprecation_Messages_and_Coverage:
     needs: [Checks, MachineLearning, Tutorials]
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8]
+        python-version: [3.9]
```

Comment:
> On thing to note, that at the point when we want to merge this PR we should update the branch rules such that 3.9 jobs, that replaced the prior 3.8 ones, are Required to pass and those old 3.8 ones are not any more.
> 
> While looking at the jobs, as we have had the new M1 jobs for some while, do we want to make them passing as Required for the merge too...

I was expecting changing python version here and in the "needs" jobs would run them with 3.9 directly, but they appear to pick up 3.8 instead. What would you suggest in this case?

Reply:
How the rules work is you select from the jobs that are run which are required - its configured. The config still has the old jobs, but since they are no longer run they show up as required but of course they will never pass since they are not run. What needs to happen is that the new jobs that we require to pass are selected/configured into the rule and the old ones removed. You can see on PRs that do not change the jobs that things are still correct since the jobs configured in the rules are the ones that are run. As this changes things we will need to update the rules. Its something we end up doing from time to time is jobs are changed, such as dropping a Python version or adding a new one.

Reply:
I've changed the required flags in `.github/actions/run-tests/action.yml`.

Reply:
The `required` flags in that file are not the branch rules I was talking about above. These should stay as they were - they indicate for each individual task CI whether it is required to pass or not and hence whether CI should should consider it failed or not. The branch rules exist in the github settings here - maybe you can view them but I doubt you can edit to update them. For each overall CI run e.g a Python 3.9 tests (i.e. each entry you see in the Build details) it configiures Github as to need the selected set to pass in order to be able to merge the PR.